.github/workflows/build.yml

```diff
@@ -117,30 +117,55 @@ jobs:
     name: "MSVC CMake (${{ matrix.build-type }}, DLL: ${{ matrix.shared-lib }})"
     runs-on: windows-latest
     env:
-      PROTOBUF_VERSION: 3.15.6
+      PROTOBUF_VERSION: 24.3
+      ABSEIL_VERSION: "20230802.0"
     steps:
       - uses: actions/checkout@v2
       - uses: ilammy/msvc-dev-cmd@v1
+        with:
+            arch: amd64
       - uses: actions/cache@v2
-        id: cache
+        id: protobuf-cache
         with:
           path: ~/protobuf-bin
           key: ${{ env.PROTOBUF_VERSION }}-${{ matrix.shared-lib }}-${{ matrix.build-type}}
+      - uses: actions/cache@v2
+        id: abseil-cache
+        with:
+          path: ~/abseil-bin
+          key: ${{ env.ABSEIL_VERSION }}-${{ matrix.shared-lib }}-${{ matrix.build-type}}
+      - name: Build and install abseil
+        if: steps.abseil-cache.outputs.cache-hit != 'true'
+        run: |
+          cd ~
+          git clone https://github.com/abseil/abseil-cpp.git -b ${{ env.ABSEIL_VERSION }} abseil
+          cd ~/abseil && mkdir build && cd build
+          cmake -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -DCMAKE_INSTALL_PREFIX=~/abseil-bin -DBUILD_SHARED_LIBS=${{ matrix.shared-lib }} -DABSL_PROPAGATE_CXX_STD=ON -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded${{ matrix.build-type == 'Debug' && 'Debug' || '' }}${{ matrix.shared-lib == 'ON' && 'DLL' || '' }} -DCMAKE_CXX_STANDARD=17 ..
+          nmake
+          nmake install
+      - name: Build and install utf8 compression algorithm
+        if: matrix.shared-lib == 'OFF'
+        run: |
+          cd ~
+          git clone https://github.com/protocolbuffers/utf8_range.git utf8_range
+          cd ~/utf8_range && mkdir build && cd build
+          cmake -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -DCMAKE_INSTALL_PREFIX=~/utf8_range-bin -DCMAKE_CXX_STANDARD=17 -Dutf8_range_ENABLE_TESTS=OFF -DBUILD_SHARED_LIBS=OFF -Dabsl_ROOT=~/abseil-bin -DCMAKE_POLICY_DEFAULT_CMP0074=NEW -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_MSVC_RUNTIME_LIBRARY='MultiThreaded${{ matrix.build-type == 'Debug' && 'Debug' || '' }}' ..
+          nmake
+          nmake install
       - name: Build and install protobuf
-        if: steps.cache.outputs.cache-hit != 'true'
+        if: steps.protobuf-cache.outputs.cache-hit != 'true'
         run: |
           cd ~
-          C:\msys64\usr\bin\wget.exe https://github.com/protocolbuffers/protobuf/releases/download/v${{ env.PROTOBUF_VERSION }}/protobuf-cpp-${{ env.PROTOBUF_VERSION }}.zip
-          7z x protobuf-cpp-${{ env.PROTOBUF_VERSION }}.zip
-          cd ~/protobuf-${{ env.PROTOBUF_VERSION }}/cmake && mkdir build && cd build
-          cmake -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX=~/protobuf-bin -Dprotobuf_BUILD_SHARED_LIBS=${{ matrix.shared-lib }} ..
+          git clone https://github.com/protocolbuffers/protobuf.git -b v${{ env.PROTOBUF_VERSION }} protobuf
```

Comment:
Should use `git clone --depth=1` here which will reduce the amount of data downloaded from ~140 MB to ~6 MB.

Reply:
agreed 👍